clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -267,7 +276,79 @@ public void testCloseOldestConnection() throws Exception {
         assertTrue("The idle connection should have been closed", selector.disconnected().containsKey(id));
         assertEquals(ChannelState.EXPIRED, selector.disconnected().get(id));
     }
+    
+    @Test
+    public void testMuteOnOOM() throws Exception {
+        //clean up default selector, replace it with one that uses a finite mem pool
+        selector.close();
+        MemoryPool pool = new SimpleMemoryPool(900, 900, false, null);
+        selector = new Selector(NetworkReceive.UNLIMITED, 5000, metrics, time, "MetricGroup", 
+                new HashMap<String, String>(), true, false, channelBuilder, pool);
+
+        try (ServerSocketChannel ss = ServerSocketChannel.open()) {
+            ss.bind(new InetSocketAddress(0));
+
+            InetSocketAddress serverAddress = (InetSocketAddress) ss.getLocalAddress();
+
+            Thread sender1 = createSender(serverAddress, randomPayload(900));
+            Thread sender2 = createSender(serverAddress, randomPayload(900));
+            sender1.start();
+            sender2.start();
+
+            //wait until everything has been flushed out to network (assuming payload size is smaller than OS buffer size)
+            //this is important because we assume both requests' prefixes (1st 4 bytes) have made it.
+            sender1.join(5000);
+            sender2.join(5000);
+
+            SocketChannel channelX = ss.accept(); //not defined if its 1 or 2
+            channelX.configureBlocking(false);
+            SocketChannel channelY = ss.accept();
+            channelY.configureBlocking(false);
+            selector.register("clientX", channelX);
+            selector.register("clientY", channelY);
+
+            selector.poll(1000);
```

Comment:
Since there is no guarantee when the server will receive those bytes, should we put this code block in a waitUntil loop?